### DISCUSSION TITLE
QMenu with separate component

### DISCUSSION THREAD

**Discussion by ctrengove** [quasarframework/quasar#11573] [General - Components / Directives / etc] [answered]
It would appear that it is not possible to have QMenu work with a target in another single file component. For example, if I have

```html
<my-custom-button>
  <q-menu>
    <q-list>
      <!-- items -->
    </q-list>
  </q-menu>
</my-custom-button>
```
and then in the file defining `<my-custom-button>` I have

```html
<template>
  <q-btn label="Label" />
</template>
```
I don't get a menu. When the button is clicked, I can see the menu being injected as a child of `<body>`, but it has no contents, the list is stripped off.

Is there any solution to this?

**Comment by pdanpdan** [comment]
Codepen?

**Comment by ctrengove** [comment]
How about this.

https://codepen.io/ctrengove/pen/XWedZbd

**Comment by pdanpdan** [comment, marked as answer]
You need to pass through the default slot, or else the menu will not have content.
https://codepen.io/pdanpdan/pen/mdBPXBz

**Reply by ctrengove** [reply]
Thanks for your help, I have it working now.
